fix(backups/healtchcheck): healthcheck must be run in task for mirror

fbeauchamp opened the pull request:
**do not squash** 

the afterBackup writer function mark the task as done, healtcheck should be  inside the run

source :  https://help.vates.tech/#ticket/zoom/21282/88085

### Description

_Short explanation of this PR (feel free to re-use commit message)_

### Checklist

- Commit
  - Title follows [commit conventions](https://bit.ly/commit-conventions)
  - Reference the relevant issue (`Fixes #007`, `See xoa-support#42`, `See https://...`)
  - If bug fix, add `Introduced by`
- Changelog
  - If visible by XOA users, add changelog entry
  - Update "Packages to release" in `CHANGELOG.unreleased.md`
- PR
  - If UI changes, add screenshots
  - If not finished or not tested, open as _Draft_

MathieuRA commented on CHANGELOG.unreleased.md:
```diff
@@ -19,6 +19,7 @@
 - [Home/VM] VMs migration from the home view will no longer execute a [Migration with Storage Motion](https://github.com/vatesfr/xen-orchestra/blob/master/docs/manage_infrastructure.md#vm-migration-with-storage-motion-vmmigrate_send) unless it is necessary [Forum#8279](https://xcp-ng.org/forum/topic/8279/getting-errors-when-migrating-4-out-5-vmguest/)(PR [#7360](https://github.com/vatesfr/xen-orc ...
 - [VM/Migration] SR is no longer required if you select a migration network (PR [#7360](https://github.com/vatesfr/xen-orchestra/pull/7360))
 - [Backup] Fix `an error has occurred` when clicking on warning text in logs (PR [#7458](https://github.com/vatesfr/xen-orchestra/pull/7458))
+- [Backup] Fix `task has already ended` when running a healt check in a mirror backup (PR [#7467](https://github.com/vatesfr/xen-orchestra/pull/7467))
```
```suggestion
- [Backup] Fix `task has already ended` when running a health check in a mirror backup (PR [#7467](https://github.com/vatesfr/xen-orchestra/pull/7467))
```

julien-f commented on @xen-orchestra/backups/_runners/_vmRunners/_AbstractRemote.mjs:
```diff
@@ -94,7 +94,6 @@ export const AbstractRemote = class AbstractRemoteVmBackupRunner extends Abstrac
     const handler = this._sourceRemoteAdapter._handler
     await Disposable.use(await handler.lock(getVmBackupDir(this._vmUuid)), async () => {
       await this._run()
-      await this._healthCheck()
```
Why isn't the `run()` method in the task?

julien-f reviewed (commented):
Instead of duplicating the `_healthCheck()` call in children, mutualize `beforeBackup()` and `afterBackup()` in the `AbstractRemote` instead.

fbeauchamp committed:
fix(backup/healthcheck): handle the case where the baseVM is not in XO cache

this can happens when the xo-server is slow to update a faster XO
xapi.getObject throw a  " no object with UUID or opaque " error during healtcheck phase

fbeauchamp committed:
fix(Backup/Mirror): task has already ended when running a health check

fbeauchamp committed:
fix(Replication/Healthcheck): use the right xapi to start healthcheck

julien-f reviewed (approved):
Fine by me, I let you rebase and merge.
